Source/pack.cpp

```diff
@@ -160,13 +159,11 @@ void __fastcall UnPackPlayer(PkPlayerStruct *pPack, int pnum, BOOLEAN killok)
 	pPlayer->_pExperience = pPack->pExperience;
 	pPlayer->_pGold = pPack->pGold;
 	pPlayer->_pMaxHPBase = pPack->pMaxHPBase;
-	v6 = pPack->pHPBase;
-	pPlayer->_pHPBase = v6;
-	if (!killok) {
-		_LOBYTE(v6) = v6 & 0xC0;
-		if (v6 < 64)
+	pPlayer->_pHPBase = pPack->pHPBase;
+	if (!killok)
+		if ((int)(pPlayer->_pHPBase & 0xFFFFFFC0) < 64)

```

Comment:
This should work (i know you tested a similar one ealier)
```suggestion
		if (pPlayer->_pHPBase >> 6 < 64)
```

Reply:
no bin exact

Reply:
or did u test this one?

Reply:
tested, only `if ((int)(pPlayer->_pHPBase & 0xFFFFFFC0) < 64)` is bin exact ((int) is necessary too , otherwise 
![image](https://user-images.githubusercontent.com/14297035/55042261-0809f280-5031-11e9-9c1b-13b7f0e17b85.png))

Reply:
Oh, it's a signed check, then this should work :)

```suggestion
		if ((int)(pPlayer->_pHPBase >> 6) < 64)
```

Reply:
but I tried it already,no! xD